gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStore.java

```diff
@@ -138,7 +139,9 @@ protected SqlStatements createSqlStatements() {
   /** Support search, unlike base class (presumably via a {@link org.apache.gobblin.runtime.api.FlowSpecSearchObject}). */
   @Override
   public Collection<Spec> getSpecsImpl(SpecSearchObject specSearchObject) throws IOException {
-    return withPreparedStatement(specSearchObject.augmentBaseGetStatement(this.sqlStatements.getStatementBase), statement -> {
```

Comment:
Is this necessary if we just trust that null (-1) start and count values wouldn't add themselves to the statement, so we wouldn't have to do this check since it would just be a sql statement without the OFFSET and LIMIT?